website_menu_by_user_status/controllers/main.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import http
+from odoo.http import request
+
+from odoo.addons.web.controllers.main import Home
+
+
+class Website(Home):
+
+    def _get_pages(self):
+        """
+        All pages width url
+        """
+        return request.env['website.menu'].sudo().search([]).filtered('url')
```

Comment:
this might be really expensive if you have a lot of menu items. Search for `[('url', '!=', False))]` should be better.

Reply:
Thank's